src/ContractInteract/EIP20Gateway.js

```diff
@@ -266,6 +266,176 @@ class EIP20Gateway {
+   * @param {string} gasPrice Gas price that staker is willing to pay for the reward.
+   * @param {string} gasLimit Maximum gas limit for reward calculation.
+   * @param {string} blockNumber Block number.
+   * @param {string} storageProof Storage proof.
+   * @param {string} hashLock Hash lock.
+   * @param {Object} txOptions Transaction option.
+   *
+   * @returns {Promise<Object>} Promise that resolves to transaction receipt.
+   */
+  confirmRedeemIntent(
+    redeemer,
+    nonce,
+    beneficiary,
+    amount,
+    gasPrice,
+    gasLimit,
+    blockNumber,
+    hashLock,
+    storageProof,
+    txOptions,
+  ) {
+    if (!txOptions) {
+      const err = new TypeError('Invalid transaction options.');
+      return Promise.reject(err);
+    }
+    return this._confirmRedeemIntentRawTx(
+      redeemer,
+      nonce,
+      beneficiary,
+      amount,
+      gasPrice,
+      gasLimit,
+      blockNumber,
+      hashLock,
+      storageProof,
+    ).then((tx) => Utils.sendTransaction(tx, txOptions));
+  }
+
+  /**
+   * Get confirm redeem intent raw transaction object
+   *
+   * @param {string} redeemer Redeemer address.
+   * @param {string} nonce Redeemer nonce.
+   * @param {string} beneficiary Beneficiary address.
+   * @param {string} amount Redeem amount.
+   * @param {string} gasPrice Gas price that staker is willing to pay for the reward.
+   * @param {string} gasLimit Maximum gas limit for reward calculation.
+   * @param {string} blockHeight Block height.
+   * @param {string} storageProof Storage proof.
+   * @param {string} hashLock Hash lock.
+   *
+   * @returns {Promise<Object>} Promise that resolves to raw transaction object.
+   */
+  _confirmRedeemIntentRawTx(
+    redeemer,
+    nonce,
+    beneficiary,
+    amount,
+    gasPrice,
+    gasLimit,
+    blockHeight,
+    hashLock,
+    storageProof,
+  ) {
+    if (!Web3.utils.isAddress(redeemer)) {
+      const err = new TypeError('Invalid redeemer address.');
+      return Promise.reject(err);
+    }
+
+    if (typeof nonce !== 'string') {
+      const err = new TypeError('Invalid nonce.');
+      return Promise.reject(err);
+    }
+
+    if (!Web3.utils.isAddress(beneficiary)) {
+      const err = new TypeError('Invalid beneficiary address.');
+      return Promise.reject(err);
+    }
+
+    if (typeof amount !== 'string') {
+      const err = new TypeError('Invalid redeem amount.');
+      return Promise.reject(err);
+    }
+
+    if (typeof gasPrice !== 'string') {
+      const err = new TypeError('Invalid gas price.');
+      return Promise.reject(err);
+    }
+
+    if (typeof gasLimit !== 'string') {
+      const err = new TypeError('Invalid gas limit.');
+      return Promise.reject(err);
+    }
+
+    if (typeof blockHeight !== 'string') {
+      const err = new TypeError('Invalid block height.');
+      return Promise.reject(err);
+    }
+
+    if (typeof storageProof !== 'string') {
+      const err = new TypeError('Invalid storage proof data.');
```

Comment:
I think it would be helpful for potential debugging to add the variables to the error strings in most cases. Generally.

Reply:
I saw later that it is done on the Facilitator. If the consumer of the mosaic.js library is not supposed to use the methods here directly, I am fine with omitting the variables, as it will be caught and printed by the `Facilitator`.

Reply:
Its already done in an extension ticket.
PR is already raised. https://github.com/OpenSTFoundation/mosaic.js/pull/71